### DISCUSSION TITLE
How to prevent specific words, commands being logged in the PSReadLine history?

### DISCUSSION THREAD

**Discussion by Marc013** [lzybkr/PSReadLine#3502] [Q&A] [answered]
I love the PSReadLine history and use of it quite often.
Whenever I do this, I see a lot of entries in this log file that I rather would not have. I'm not interested when I've used `cls` or a custom function I've created for opening the PSReadLine history file.

Is there a way I can define a list of words, commands that should not be logged?

**Comment by daxian-dbw** [comment, marked as answer]
You can use `Set-PSReadLineOption -AddToHistoryHandler` to do the filtering. See https://learn.microsoft.com/en-us/powershell/module/psreadline/set-psreadlineoption?view=powershell-7.3#-addtohistoryhandler for more details.

**Reply by Marc013** [reply]
Works like a charm!
Thank you for providing the solution.

I've added the following to my pwsh profile

```PowerShell
Set-PSReadLineOption -AddToHistoryHandler {
    param([string]$line)

    $exclude = 'historyFile|cls|ls'
    return ($line -notmatch $exclude)
}
```

I found this code snipped at [https://megamorf.gitlab.io/cheat-sheets/powershell-psreadline/](https://megamorf.gitlab.io/cheat-sheets/powershell-psreadline/)
